solutionbox/structured_data/mltoolbox/_structured_data/_package.py

```diff
@@ -175,47 +156,69 @@ def cloud_analysis(output_dir, dataset, project_id=None):
     raise ValueError('CsvDataSet should be built with a file pattern, not a '
                      'list of files.')
 
-  _assert_gcs_files([output_dir, dataset.input_files[0]])
+  if project_id and not cloud:
+    raise ValueError('project_id only needed if cloud is True')
+
+  if cloud:
+    _assert_gcs_files([output_dir, dataset.input_files[0]])
+
 
-  # Write schema to a file.
   tmp_dir = tempfile.mkdtemp()
-  _, schema_file_path = tempfile.mkstemp(dir=tmp_dir, suffix='.json',
-                                        prefix='schema')
   try:
+    # write the schema file.
+    _, schema_file_path = tempfile.mkstemp(dir=tmp_dir, suffix='.json',
+                                           prefix='schema')    
     file_io.write_string_to_file(schema_file_path, json.dumps(dataset.schema))
 
-    args = ['cloud_preprocess',
+    args = ['preprocess',
             '--input-file-pattern=%s' % dataset.input_files[0],
             '--output-dir=%s' % output_dir,
             '--schema-file=%s' % schema_file_path]
 
-
-    print('Starting cloud preprocessing.')
-    print('Track BigQuery status at')
-    print('https://bigquery.cloud.google.com/queries/%s' % _default_project())
-    preprocess.cloud_preprocess.main(args)
-    print('Cloud preprocessing done.')
+    if cloud:
+      print('Starting cloud preprocessing.')
+      print('Track BigQuery status at')
+      print('https://bigquery.cloud.google.com/queries/%s' % _default_project())
+      preprocess.cloud_preprocess.main(args)
+      print('Cloud preprocessing done.')
+    else:
+      print('Starting local preprocessing.')
```

Comment:
done. Kept it on online prediction because it does not return a job (it returns pandas df)